orbit/pkg/table/find_cmd/find_cmd_darwin.go

```diff
@@ -0,0 +1,146 @@
+	"fmt"
+	"io"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/rs/zerolog/log"
+)
+
+// Columns is the schema of the table.
+func Columns() []table.ColumnDefinition {
+	return []table.ColumnDefinition{
+		// directory is the first argument of the find command (basically
+		// where to search for files).
+		table.TextColumn("directory"),
+		// type allows setting find's '-type' argument.
+		table.TextColumn("type"),
+		// perm allows setting find's '-perm' argument.
+		table.TextColumn("perm"),
+		// path are the found directories.
+		table.TextColumn("path"),
+	}
+}
+
+var permRegexp = regexp.MustCompile("[-+]*\\d+")
+
+// Generate is called to return the results for the table at query time.
+//
+// Constraints for generating can be retrieved from the queryContext.
+func Generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	getArgumentOpEqual := func(argName string) string {
+		argValue := ""
+		if constraints, ok := queryContext.Constraints[argName]; ok {
+			for _, constraint := range constraints.Constraints {
+				if constraint.Operator == table.OperatorEquals {
+					argValue = constraint.Expression
+				}
+			}
+		}
+		return argValue
+	}
+
+	directory := getArgumentOpEqual("directory")
+	if directory == "" {
+		return nil, errors.New("missing directory argument")
+	}
+	if !filepath.IsAbs(directory) {
+		return nil, errors.New("directory must be an absolute path")
+	}
+
+	findType := getArgumentOpEqual("type")
+	if findType != "" {
+		switch findType {
+		case "b", "c", "d", "f", "l", "p", "s":
+			// OK
+		default:
+			return nil, errors.New("type must be one of: 'b', 'c', 'd', 'f', 'l', 'p' or 's'")
+		}
+	}
+
+	perm := getArgumentOpEqual("perm")
+	if perm != "" {
+		if !permRegexp.Match([]byte(perm)) {
+			return nil, fmt.Errorf("perm must be of the form: %s", permRegexp)
+		}
+	}
+
+	args := []string{directory}
+	if findType != "" {
+		args = append(args, "-type", findType)
+	}
+	if perm != "" {
+		args = append(args, "-perm", perm)
+	}
+
+	cmd := exec.Command("/usr/bin/find", args...)
+	stdoutPipe, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, fmt.Errorf("create stdout pipe: %w", err)
+	}
+	stderrPipe, err := cmd.StderrPipe()
+	if err != nil {
+		return nil, fmt.Errorf("create stderr pipe: %w", err)
+	}
+
+	if err := cmd.Start(); err != nil {
+		return nil, fmt.Errorf("command start failed: %w", err)
+	}
+
+	var outDirs []string
+	reader := bufio.NewReader(stdoutPipe)
+	line, err := reader.ReadString('\n')
+	for err == nil {
+		line = strings.TrimSuffix(line, "\n")
+		if line == "" {
+			continue
+		}
+		outDirs = append(outDirs, line)
+		line, err = reader.ReadString('\n')
+	}
+	if err != io.EOF {
+		return nil, fmt.Errorf("unexpected error: %w", err)
+	}
+
+	stderr, err := io.ReadAll(stderrPipe)
+	if err != nil {
+		log.Debug().Err(err).Msg("failed to read find stderr")
+	}
+
+	if err := cmd.Wait(); err != nil {
```

Comment:
Shouldn't we do this wait BEFORE we start reading from the reader (line 112)?

Reply:
It could be that by the nature of `reader.ReadString` it will wait for the cmd. 
But not sure it the EOF is not there if the cmd is too slow.

Not sure hence why maybe the wait should come first 🤷‍♂️  ?

Reply:
See https://pkg.go.dev/os/exec#Cmd.StdoutPipe
```
[...] It is thus incorrect to call Wait before all reads from the pipe have completed. For the same reason, it is incorrect to call Run when using StdoutPipe. [...]
```

Reply:
Thanks for clarifying.